dev/Kozani/KozaniRemoteManager/KozaniDvcServer.cpp

```diff
@@ -0,0 +1,734 @@
+                    }
+                }
+                break;
+
+            case Dvc::ActivationKind::File:
+                // ToDo: support FTA launch.
+                break;
+
+            case Dvc::ActivationKind::Protocol:
+                // ToDo: support protocol launch.
+                break;
+
+            default:
+                FAIL_FAST_MSG("Logic error - should have handled all supported launch contracts.");
+                break;
+        }
+        
+        // When we reach here, activation is successful.
+        bool isNewInstance{};
+        {
+            auto lock{ g_connectionManager.m_activityMapLock.lock_exclusive() };
+            RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), g_connectionManager.m_closing);
+
+            auto iter{ g_connectionManager.m_activityMap.find(pdu.activity_id()) };
+            if (iter == g_connectionManager.m_activityMap.end())
+            {
+                // Rare case - the app is terminated right after launch and the activity has been removed from the map from the other thread tracking 
+                // the lifetime of the process, before reaching here. There is nothing to do after that.
+                return S_OK;
+            }
+
+            AppActivationInfo& appInfo{ iter->second };
+            if (appInfo.activationStatus == AppActivationStatus::TerminationRequestedFromClient)
+            {
+                // The client has already requested termination of the app. Honor it now. 
+                wil::unique_handle process{ OpenProcess(PROCESS_TERMINATE, FALSE, processId) };
+                if (process)
+                {
+                    LOG_IF_WIN32_BOOL_FALSE_MSG(TerminateProcess(process.get(), 0), "Failed to terminate process %u", processId);
+                }
+                else
+                {
+                    LOG_LAST_ERROR_MSG("Failed to open process %u with PROCESS_TERMINATE access right.", processId);
+                }
+
+                // Remove the activity entry from the m_activityMap as we no longer track it.
+                g_connectionManager.m_activityMap.erase(iter);
+                return S_OK;
+            }
+
+            auto pidAdded{ g_connectionManager.m_processIdMap.emplace(processId, pdu.activity_id()) };
+            if (pidAdded.second)
+            {
+                // New process Id in the tracking map. The activation creates a new process.
+                isNewInstance = true;
+                appInfo.pid = processId;
+
+                // Track lifetime of the process.
+                appInfo.processHandle.reset(OpenProcess(SYNCHRONIZE | PROCESS_TERMINATE, FALSE, processId));
+                if (!appInfo.processHandle)
+                {
+                    g_connectionManager.m_activityMap.erase(iter);
+
+                    errorMessage = "Failed to OpenProcess after the app is activated. The app may have crashed. Process Id: " + std::to_string(processId);
+                    RETURN_LAST_ERROR();
```

Comment:
Always risky to wait to retrieve last error. Also just from a loging / troubleshooting / analysis / timing perspective you want to log a failure as immediately as possible, before doing additional work.

SUGGESTION:

```
appInfo.processHandle = std::move(OpenProcess(SYNCHRONIZE | PROCESS_TERMINATE, FALSE, processId);
if (!appInfo.processHandle)
{
    const auto hr{ LOG_LAST_ERROR_MSG("ProcessId:%u", processId) };
    g_connectionManager.m_activityMap.erase(iter);
    return hr;
}
```

Reply:
Added 
const auto hrOpenProcess{ LOG_LAST_ERROR_MSG("ProcessId: %u", processId) };
before 
m_activityMap.erase(iter);